virttest/utils_net.py

```diff
@@ -1486,6 +1486,38 @@ def set_guest_ip_addr(session, mac, ip_addr,
         raise IPAddrSetError(mac, ip_addr, err)
 
 
+def get_guest_default_gateway(session):
+    """
+    Get guest gateway from serial session for linux guest
+    :param session: serial session
+    :return: return the default gateway
+    """
+    cmd = "ip route | awk '/default/ { print $3 }'"
+    try:
+        output = session.cmd_output(cmd).strip()
+        logging.debug("Guest default gateway is %s" % output)
+    except process.CmdError:
+        logging.error("Failed to get the guest's default Gateway")
+        return None
+    return output
+
+
+def get_guest_nameserver(session):
+    """
+    Get guest nameserver from serial session for linux guest
+    :param session: serial session
+    :return: return nameserver
+    """
+    cmd = "cat /etc/resolv.conf  | grep nameserver"
+    try:
+        output = session.cmd_output(cmd).strip()
+        logging.debug("Guest name server is %s" % output)
+    except process.CmdError:
```

Comment:
same here

Reply:
updated as above

Reply:
Thank you for your suggestion, updated as below: 
    output = None
    try:
        output = session.cmd_output(cmd).strip()
        logging.debug("Guest name server is %s" % output)
    except (aexpect.ShellError, aexpect.ShellTimeoutError):
        logging.error("Failed to get the guest's nameserver")
    return output